agent/consul/client.go

```diff
@@ -251,6 +259,10 @@ func (c *Client) RPC(method string, args interface{}, reply interface{}) error {
 	if err := c.connPool.RPC(c.config.Datacenter, server.Addr, server.Version, method, server.UseTLS, args, reply); err != nil {
 		c.routers.NotifyFailedServer(server)
 		c.logger.Printf("[ERR] consul: RPC failed to server %s: %v", server.Addr, err)
+		if retry && structs.IsErrNoLeader(err) {
+			retry = false
```

Comment:
Is it worth adding a small backoff period here and more than one retry instead of just one?

Reply:
The server manager shifts away from this server when you call `NotifyFailedServer` so this should be safe without an additional delay.

Reply:
Maybe I'll put a 1 second delay in there, just for good measure (I don't think we need more than 1 retry though).

Reply:
After thinking on this more I'm going to take your suggestion and apply the RPC hold timeout here as well - this will simplify things.